### ISSUE TITLE
[Bug]: `react/no-unknown-property` incorrectly flags `closedby` on `<dialog>`

### ISSUE THREAD

**Issue by jolacdev** [jsx-eslint/eslint-plugin-react#3980]
### Is there an existing issue for this?

- [x] I have searched the existing issues and my issue is unique
- [x] My issue appears in the command-line and not only in the text editor

### Issue

The rule `react/no-unknown-property` flags the `closedby` attribute as invalid when used on the native `<dialog>` element, even though `closedby` is a valid HTML attribute for `<dialog>`.

```jsx
export const Dialog = () => (
  <dialog closedby="any">
    <p>Test</p>
  </dialog>
);
```

ESLint reports: _Unknown property 'closedby' found eslint(react/no-unknown-property)_

### Possible fix

Add `closedby` to the allowed attributes for `<dialog>` in `no-unknown-property.js`, e.g.:
```js
const ATTRIBUTE_TAGS_MAP = {
...
  onClose: ['dialog'],
  closedby: ['dialog'],
...
}
```

### Related
- Issue: #3436

### Expected Behavior

No error.

### eslint-plugin-react version

v7.37.5

### eslint version

v9.36.0

### node version

v24.13.0
